docs/fabric-native-components.md

```diff
@@ -0,0 +1,188 @@
+
+This guide will show you how to build Fabric Native Component, by implementing a simple centered text component. The steps to doing this are:
+
+1. Define a JavaScript specification using Flow or TypeScript.
+2. Implement the Native code.
+3. Use the Component in an App.
+
+# Creating the Fabric Native Centered Text Component
+
+You're going to need a simple application to use the component:
+
+```bash
+npx @react-native-community/cli@latest init Demo --install-pods false
+```
+
+Next you'll need to create a folders structure to hold our `RTNCenteredText` component:
+
+```bash
+mkdir -p Demo/js
+```
+
+This gives you the following layout:
+
+```
+Demo
+├── android
+├── ios
+└── js
+```
+
+## 1. Define Specification for Codegen
+
+Your specification must be defined in either [TypeScript](https://www.typescriptlang.org/) or [Flow](https://flow.org/) (see [Codegen](the-new-architecture/what-is-codegen) documentation for more details). This is used by Codegen to generate the C++, Objective-C++ and Java to connect your platform code to the JavaScript runtime that React runs in.
+
+Use this specification for our Text Component:
+
+<Tabs groupId="language" queryString defaultValue={constants.defaultJavaScriptSpecLanguage} values={constants.javaScriptSpecLanguages}>
+<TabItem value="typescript">
+
+The specification file must be named `<MODULE_NAME>NativeComponent.{ts,tsx}` to work with Codegen:
+
+```typescript title="Demo/js/RTNCenteredTextNativeComponent.ts"
+import type {ViewProps} from 'react-native';
+import type {HostComponent} from 'react-native/Libraries/Renderer/shims/ReactNativeTypes';
+import type {Int32} from 'react-native/libraries/Types/CodegenTypes';
+import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
+
+export interface NativeProps extends ViewProps {
+  text?: string;
+  // You can add other props here
+}
+
+export default codegenNativeComponent<NativeProps>(
+  'RTNCenteredText',
+) as HostComponent<NativeProps>;
+```
+
+</TabItem>
+<TabItem value="flow">
+The specification file must be named `<MODULE_NAME>NativeComponent.{js,jsx}` to work with Codegen:
+
+```flow title="Demo/js/RTNCenteredTextNativeComponent.js":
+// @flow strict-local
+
+import type {ViewProps} from 'react-native/Libraries/Components/View/ViewPropTypes';
+import type {HostComponent} from 'react-native';
+import type {Int32} from 'react-native/libraries/Types/CodegenTypes';
+import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
+
+type NativeProps = $ReadOnly<{|
+  ...ViewProps,
+  text: ?string,
+  margin?: Int32,
```

Comment:
the flow specs does not have the margin anymore. I think we should remove them from here as well and we should also remove the import of the Int32